price_calculator/src/main/kotlin/pl/edu/pg/rsww/price_calculator/PriceCalculator.kt

```diff
@@ -0,0 +1,48 @@
+package pl.edu.pg.rsww.pricecalculator
+
+import kotlin.math.PI
+import kotlin.math.acos
+import kotlin.math.cos
+import kotlin.math.round
+import kotlin.math.sin
+
+public class PriceCalculator {
+    val earthRadius = 6371 // in km
+
+    fun toRadians(deg: Float): Double = deg / 180.0 * PI
+
+    fun calculateDistance(
+        location1: Geolocation,
+        location2: Geolocation,
+    ): Int {
+        // acos(sin(lat1) * sin(lat2) + cos(lat1) * cos(lat2) * cos(lon2-lon1)) * 6371 = acos(a + b * c) * 6371
+        // c = cos(lon2-lon1)
+        val a = sin(toRadians(location1.latitude)) * sin(toRadians(location2.latitude))
+        val b = cos(toRadians(location1.latitude)) * cos(toRadians(location2.latitude))
+        val c = cos(toRadians(location2.longitude - location1.longitude))
+        val result = acos(a + b * c) * 6371
+        val ret = round(result).toInt()
+        return ret
+    }
+
+    fun calculatePrice(
```

Comment:
Could we include the selected meal option in this calculation? It will probably be a bit surprising if the price doesn't change when e.g. switching from no meals to all-inclusive. I think it would be completely fine to either hardcode the prices for all of the 10 available meal options or adding a fixed price of the selected meal option to the dataset and then using that as input to this service. Whichever solution were to be selected, some price would need to be selected for all of these: https://github.com/YetAnotherSpieskowcy/travel-agency-samples-mongo/blob/main/61_meals.js

*One could also argue there's a similar argument to be made for rooms as well but we'd have to completely randomly generate some price data for that as it's impossible to get through hundreds of different room types that we have in the data set.*